modules/gapi/src/streaming/vpl/vpl_legacy_source_engine.cpp

```diff
@@ -0,0 +1,321 @@
+#include <algorithm>
+#include <exception>
+
+#include "streaming/vpl/vpl_legacy_source_engine.hpp"
+#include "streaming/vpl/vpl_utils.hpp"
+#include "streaming/vpl/vpl_dx11_accel.hpp"
+#include "streaming/vpl/surface/surface.hpp"
+
+#include "logger.hpp"
+
+
+namespace cv {
+namespace gapi {
+namespace wip {
+/* UTILS */
+mfxU32 GetSurfaceSize(mfxU32 FourCC, mfxU32 width, mfxU32 height) {
+    mfxU32 nbytes = 0;
+
+    switch (FourCC) {
+        case MFX_FOURCC_I420:
+        case MFX_FOURCC_NV12:
+            nbytes = width * height + (width >> 1) * (height >> 1) + (width >> 1) * (height >> 1);
```

Comment:
I don't think we would get much performance of it, so I'd make it more readable and change to `/ 2`